src/common/window-utils.ts

```diff
@@ -25,7 +25,7 @@ export class WindowUtils {
     }
 
     public setTimeout(handler: Function, timeout: number): number {
-        return window.setTimeout(handler, timeout);
+        return globalThis.setTimeout(handler, timeout);
```

Comment:
It feels a little off to be setting up `WindowUtils` to be only partially compatible with a `window`-free enviornment like this. The only place the service worker uses it is to pass it via `TabContextFactory` to `InjectorController`; perhaps it might make more sense to replace the `WindowUtils` dependency in those two classes with a `setTimeout` dependency and just pass `globalThis.setTimeout` from `service-worker-init`, to avoid having to create `WindowUtils` there at all?

Reply:
oh good call! I like this solution.